luxtronik/__init__.py

```diff
@@ -63,9 +63,23 @@ def __init__(self, host, port=LUXTRONIK_DEFAULT_PORT, safe=True):
         self._port = port
         self._safe = safe
         self._socket = None
-        self.read()
+        self._connect()
 
     def __del__(self):
+        self._disconnect()
+
+    def _connect(self):
```

Comment:
I am not sure, but should 'self._lock' also used in the `_connect` method?

Reply:
(and the same for `_disconnect`?)

Reply:
`_connect` is called inside `_lock_and_connect` for for multithreaded access to the instance. `__init__` and `__del__` are only called by one thread.